src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs

```diff
@@ -136,14 +136,14 @@ End Sub
                 var solutionChecksum = await solution.State.GetChecksumAsync(CancellationToken.None);
 
                 var source = new CancellationTokenSource();
-                var connection = new InvokeThrowsCancellationConnection(source);
+                var connection = new OwnedDisposable<RemoteHostClient.Connection>(new InvokeThrowsCancellationConnection(source)).Box();
                 var exception = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => SessionWithSolution.CreateAsync(connection, solution, source.Token));
                 Assert.Equal(exception.CancellationToken, source.Token);
 
                 // make sure things that should have been cleaned up are cleaned up
                 var service = (RemotableDataServiceFactory.Service)solution.Workspace.Services.GetService<IRemotableDataService>();
                 Assert.Null(service.GetRemotableData_TestOnly(solutionChecksum, CancellationToken.None));
-                Assert.True(connection.Disposed);
+                Assert.False(((InvokeThrowsCancellationConnection)connection.Target).Disposed);
```

Comment:
@heejaechang This type of situation is what led me to make the comment here: https://github.com/dotnet/roslyn/pull/26178#discussion_r182267145.

Reply:
still not sure what you mean. original code was as soon as connection is given to SessionWithSolution.CreateAsync ownership is moved according to your term. and CreateAsync got the responsibility to dispose it.

now it looks like you changed that to not move ownership until the constructor which is after test is designed to throw cancellation. so sure the test will fail. since this PR changed responsibility of disposing the connection to the extension method. previous PR explicitly tried to move the responsibility to the CreateAsync. and this basically reverted that.

(still not sure why there is Box, and then get the resource from it, rather than pass down ref Owner... thing to CreateAsync)

anyway, you can make this test to do SessionWithSolution.CreateAsync(connection.Move(), ..) to have same semantic if you want to.

Reply:
I like your new pattern better though, connection can always dispose his connection without worrying about ownership. making ownership tracking simpler. but can't be done by your refcountingdisposable type as well.